tests/components/roomba/test_config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Test the iRobot Roomba config flow."""
+from asynctest import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.roomba.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.roomba.const import DOMAIN
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.roomba.config_flow.PlaceholderHub.authenticate",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.roomba.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.roomba.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Name of the device"
+    assert result2["data"] == {
+        "host": "1.1.1.1",
+        "username": "test-username",
+        "password": "test-password",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roomba.config_flow.PlaceholderHub.authenticate",
```

Comment:
This needs to patch `roombapy`

Reply:
Unfortunately, we can't merge config flows without tests.

You'll need to do something similar to the below:


Passing
```
mocked_roomba = _create_mocked_roomba()

with patch("homeassistant.components.roomba.config_flow.Roomba",return_value=mocked_roomba)
```

Failing
```
mocked_roomba = _create_mocked_roomba(connect=RoombaConnectionError)

with patch("homeassistant.components.roomba.config_flow.Roomba",return_value=mocked_roomba)
```

```
def _create_mocked_roomba(connect=None):
  mocked_roomba = MagicMock()
  type(mocked_roomba).connect = MagicMock(side_effect=connect)
  return mocked_roomba
```

Reply:
Gentle reminder that we do need tests for the config flow to be able to merge.